pages/security/csp.md

```diff
@@ -0,0 +1,115 @@
+sources, and a usable sample policy could look something like this:
+
+`"default-src 'self'; srcipt-src https://facebook.com; child-src https://youtu.be; object-src 'none';"`
+
+Let's break this down:
+
+```
+  default-src self : Only load content from the current
+    origin, exluding subdomains
+
+  script-src https://facebook.com : Allow
+    scripts (e.g. `Like` button code) from Facebook
+    to load over HTTPS
+
+  child-src https://youtu.be : Allow embedded content
+    from youtube over HTTPS
+
+  object-src none : Prevent flash content from loading
+```
+
+A full list of available directives and keywords is outside the scope of this guide, but is available in the excellent Google CSP guide linked in the [Further Reading](#further-reading) section.
+
+It might be useful to test your policies before letting them use on your users. To do this, use the `Content-Security-Policy-Report-Only` HTTP header. Combined with the reporting information in the next section, you can monitor the kinds of content your user's are encountering and tweak the your policy accordingly.
+
+
+#### Reporting
+CSP can also be configured to send reports to an endpoint you control when content that violates your policies is encountered.
+
+To do this, use the `report-uri` directive, passing it a fully qualified URI e.g. `https://my-public-site.gov/reports/csp`. Now, each time content from an invalid source is encountered, your browser will POST a JSON payload to the provided URI.
+
+Here is an example, from the MDN article linked in [Further Reading](#further-reader)
+
+```
+{
+  "csp-report": {
+    "document-uri": "http://example.com/signup.html",
+    "referrer": "",
+    "blocked-uri": "http://example.com/css/style.css",
+    "violated-directive": "style-src cdn.example.com",
+    "original-policy": "default-src 'none'; style-src cdn.example.com; report-uri /_/csp-reports"
+  }
+}
+```
+
+This data can of course be mixed with other request metadata, like IP address or browser User-Agent, and fed into whatever log processing system your app has in place.
+
+Reporting can only be enabled via HTTP header, not inside a `<meta>` tag!
+
+
+
+### Client-side Implementation
+To implement CSP on the client, add a `<meta>` tag to your web site's `<head>`.
+
+`<meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src https://*; child-src 'none';">`
+
+You're done!
+
+### Server-side Implementation
+Similarly, to include CSP on the server, simply return a `Content-Security-Policy: {my-policy-string}` HTTP header in your web responses.
+
+<br>
+
+Although a basic CSP policy is relatively easy to write, you might find it easier to use a server-side library to define your policies declaratively. Two of well-known and actively maintained libraries are [secureheaders](https://github.com/twitter/secureheaders), a **ruby** gem written and maintained by Twitter, and the [helmetjs](https://github.com/helmetjs/helmet) middleware for **express.js**.
```

Comment:
A lot of 18F projects use Django, and [django-cors-headers](https://github.com/ottoyiu/django-cors-headers) is the well-maintained, go-to solution for CORS on it. You're welcome to mention it too if you want, no worries if not!

Reply:
Oh geez I just confused CORS with CSP. 😣 

The go-to CSP package is generally [django-csp](https://github.com/mozilla/django-csp).